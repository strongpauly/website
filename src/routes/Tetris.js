import React, { Component } from 'react';
import {HiFrame} from '../components/HiFrame';

export class Tetris extends Component {

  render() {
    return <div id="tetrisContainer">
      <p>
        The classic Gameboy game implemented with react and redux.
        It&apos;s source can be found <a href="https://github.com/strongpauly/tetris">here</a>.
      </p>
      <HiFrame src="/tetris" title="tetris"></HiFrame>
    </div>;
  }

}
