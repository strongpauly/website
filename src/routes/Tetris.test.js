import React from 'react';
import Tetris from './Tetris';
import {shallow} from 'enzyme';

/* eslint-env jest */

describe('Tetris', () => {
  it('renders without crashing', () => {
    const app = shallow(<Tetris/>);
    expect(app).toMatchSnapshot();
  });

  it('creates a #tetrisContainer', () => {
    const app = shallow(<Tetris/>);
    expect(app.find('#tetrisContainer')).toHaveLength(1);
  });
});
