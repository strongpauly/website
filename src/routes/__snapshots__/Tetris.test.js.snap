// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tetris renders without crashing 1`] = `
<div
  id="tetrisContainer"
>
  <div
    className="container"
  >
    The classic Gameboy game implemented with react and redux. It's source can be found 
    <a
      href="https://github.com/strongpauly/tetris"
    >
      here
    </a>
    .
  </div>
  <HiFrame
    src="/tetris"
    title="tetris"
  />
</div>
`;
